script/vcnl4010/vcnl4010.lua

```diff
@@ -90,120 +109,174 @@ function vcnl4010:_get(reg, pack_fmt, convert)
     return string.unpack(pack_fmt, buffer)
 end
 
+function vcnl4010:_update(reg, pack_fmt, update)
+    -- read register, apply a function to the result and write it back
+    -- number of bytes to read based on the pack string
+    if not update then
+        assert(false, "you must supply an update function")
+    end
+    local pack_size = string.packsize(pack_fmt)
+    local status, buffer =
+        i2c.txn(i2c.tx(self.address, reg), i2c.rx(self.address, pack_size))
+    if not status then
+        return false, "failed to get value from device"
+    end
+    -- call update function
+    local newvalue = string.unpack(pack_fmt, buffer)
+    newvalue = update(newvalue)
+    status, buffer =
+        i2c.txn(i2c.tx(self.address, reg, string.pack(pack_fmt, newvalue)))
+    if not status then
+        return false, "unable to set value"
+    end
+    return newvalue
+end
+
 function vcnl4010:set_led_current(current)
     -- set the LED current, table 4
     if current < 0 or current > 20 then
-        return false, "VCNL4010 LED current value should be between 0 and 20"
+        assert(false, "VCNL4010 LED current value should be between 0 and 20")
```

Comment:
You can omit the whole if statement.. 

`assert(current < 0 or current > 20, "...")`